ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala

```diff
@@ -69,25 +69,31 @@ final class AuthorizationInterceptor(
 
   private[this] def resolveAuthenticatedUserRights(claimSet: ClaimSet): Future[ClaimSet] =
     claimSet match {
-      case ClaimSet.AuthenticatedUser(userId, participantId, expiration) =>
-        userManagementService
-          .listUserRights(Ref.UserId.assertFromString(userId))
-          .map {
-            case Left(msg) =>
-              logger.warn(
-                s"Authorization error: cannot resolve rights for user '$userId' due to $msg."
-              )
-              ClaimSet.Unauthenticated
-            case Right(userClaims) =>
-              ClaimSet.Claims(
-                claims = userClaims.view.map(userRightToClaim).toList.prepended(ClaimPublic),
-                ledgerId = None,
-                participantId = participantId,
-                applicationId = Some(userId),
-                expiration = expiration,
-                resolvedFromUser = true,
-              )
-          }
+      case ClaimSet.AuthenticatedUser(userIdStr, participantId, expiration) =>
+        Ref.UserId.fromString(userIdStr) match {
+          case Left(err) =>
+            logger.warn(s"Authorization error: invalid token because $err")
```

Comment:
Should we provide that information to the client? Receiving `Unauthenticated` without any context on why your token was invalid seems potentially hard to debug. From a security pov, I don’t see an issue with exposing that info.

Reply:
Good point. However, in case the client is also the admin of the node this is not a problem.

In case, the client is no admin of the node, the `UNAUTHENTICATED` here is similar to the `UNAUTHENTICATED` of non-existing user isn't it? It would in particular fail anyways later on the user-lookup. Would you expect the client to learn that the user in his token does not exist?

Reply:
I think there is a difference here: The format of a user id is public knowledge. I could check client side before I send the request if my user id is valid.

The list of users that have been created is not public knowledge on the other hand.

So I’m not asking to differentiate between user exists or not but between correct or incorrect user id purely based on the format.

Reply:
You've convinced me. I've adjusted the code accordingly.